### PR TITLE
Regenerate animals around the player.

### PR DESCRIPTION
**Pull request by modmuss50**
Very quick and dirty, but should be good enough. Better ideas are welcome.

### CODE REVIEW

File: mod/src/main/java/holiday/server/CommonEntrypoint.java [hunk 1]
```diff
@@ -58,6 +87,52 @@ public void onInitialize() {
...
+                return;
+            }
+
+            if (world.getRegistryKey() != World.OVERWORLD) {
+                return;
+            }
+
+            for (ServerPlayerEntity player : world.getPlayers()) {
+                // We only need to fix chunks within the initial world border
+                if (player.getX() < -CHUNK_RESPAWN_RADIUS_BLOCKS || player.getX() > CHUNK_RESPAWN_RADIUS_BLOCKS || player.getZ() < -CHUNK_RESPAWN_RADIUS_BLOCKS || player.getZ()  > CHUNK_RESPAWN_RADIUS_BLOCKS) {
```
Comment (Octol1ttle) [inline mod/src/main/java/holiday/server/CommonEntrypoint.java:102]:
```suggestion
                if (player.getX() < -CHUNK_RESPAWN_RADIUS_BLOCKS || player.getX() > CHUNK_RESPAWN_RADIUS_BLOCKS || player.getZ() < -CHUNK_RESPAWN_RADIUS_BLOCKS || player.getZ() > CHUNK_RESPAWN_RADIUS_BLOCKS) {
```

File: mod/src/main/java/holiday/server/CommonEntrypoint.java [hunk 2]
```diff
@@ -58,6 +87,52 @@ public void onInitialize() {
...
+                    return;
+                }
+
+                for (int x = -SCAN_RADIUS_CHUNKS; x < SCAN_RADIUS_CHUNKS; x++) {
+                    for (int z = -SCAN_RADIUS_CHUNKS; z < SCAN_RADIUS_CHUNKS; z++) {
```
Comment (Octol1ttle) [inline mod/src/main/java/holiday/server/CommonEntrypoint.java:107]:
```suggestion
                for (int x = -SCAN_RADIUS_CHUNKS; x <= SCAN_RADIUS_CHUNKS; x++) {
                    for (int z = -SCAN_RADIUS_CHUNKS; z <= SCAN_RADIUS_CHUNKS; z++) {
```

File: mod/src/main/java/holiday/server/CommonEntrypoint.java [hunk 3]
```diff
@@ -58,6 +87,52 @@ public void onInitialize() {
...
+
+                        if (!CHECKED_CHUNKS.add(chunkLongPos)) {
+                            continue;
+                        }
+
+                        ChunkPos chunkPos = new ChunkPos(chunkLongPos);
+                        WorldChunk chunk = world.getChunk(chunkPos.x, chunkPos.z);
+
+                        if (Boolean.TRUE.equals(chunk.getAttached(ANIMALS_REGENERATED_CHUNK_TYPE))) {
+                            return;
```
Comment (Octol1ttle) [inline mod/src/main/java/holiday/server/CommonEntrypoint.java:122]:
```suggestion
                            continue;
```

**Review by Octol1ttle — CHANGES_REQUESTED** [review]

File: mod/src/main/java/holiday/server/CommonEntrypoint.java [hunk 4]
```diff
@@ -58,6 +87,52 @@ public void onInitialize() {
...
+            }
+
+            if (world.getRegistryKey() != World.OVERWORLD) {
+                return;
+            }
+
+            for (ServerPlayerEntity player : world.getPlayers()) {
+                // We only need to fix chunks within the initial world border
+                if (player.getX() < -CHUNK_RESPAWN_RADIUS_BLOCKS || player.getX() > CHUNK_RESPAWN_RADIUS_BLOCKS || player.getZ() < -CHUNK_RESPAWN_RADIUS_BLOCKS || player.getZ() > CHUNK_RESPAWN_RADIUS_BLOCKS) {
+                    return;
```
Comment (Octol1ttle) [inline mod/src/main/java/holiday/server/CommonEntrypoint.java:103]: missed this one
```suggestion
                    continue;
```

**Comment by modmuss50** [comment]
Tested this on a backup of the world and seems to be working.

**Pull request merged**